client/src/components/kvEditor/KVEditorRow.tsx

```diff
@@ -157,17 +158,28 @@ function KVEditorRow({
 
   const eventHandlers = EditorView.domEventHandlers({
     paste(event) {
-      event.preventDefault();
-      if (!event.clipboardData) {
+      if (!event.target || !event.clipboardData) {
+        return;
+      }
+      // differenciates between initial and synthetic event to prevent infinite loop
+      if (!event.isTrusted) {
         return;
       }
-      const text = event.clipboardData.getData('text');
+      event.preventDefault();
+
+      const text = event.clipboardData.getData('text/plain');
       const sanitized = text.replace(/(\r\n|\n|\r)/gm, '');
-      const selection = window.getSelection();
-      if (!selection || !selection.rangeCount) return;
-      selection.deleteFromDocument();
-      selection.getRangeAt(0).insertNode(document.createTextNode(sanitized));
-      selection.collapseToEnd();
+
+      const data = new DataTransfer();
+      data.setData('text/plain', sanitized);
+      const sanitizedEvent = new ClipboardEvent('paste', {
+        bubbles: true,
```

Comment:
yes I like this much better. Can we do `{ ...event, clipboardData: data }` here so that we don't accidentally forget anything.

Reply:
Yea, I tried that, too, but some property hindered dispatching to work. But I can try to single out the property that caused it and set it correctly.

Reply:
Spreading the event object into the constructor doesn't work. The constructor expect an object of type `EventInit` or in our case `ClipBoardEventInit`, which differs from `Event`. 
`EventInit` only contains 
```
interface EventInit {
    bubbles?: boolean;
    cancelable?: boolean;
    composed?: boolean;
}
```
Many (all?) of the other props are read only and set by the browser on dispatch.

I guess it makes sense to leave it as is then. The prop values that are currently set for the new event, match the ones of the initial eEvent.

Reply:
OK then don't worry, just go with the hardcoded values.